### ISSUE TITLE
Unzipping content fails under certain conditions

### ISSUE THREAD

**Issue by sebawern** [microsoft/Microsoft-Win32-Content-Prep-Tool#2]
Unzipping the content fails on clients when there are not only files but also folders (and nested folders) in the package.

My workaround is a powershell script that manually creates the required folder, moves all files to their original location and calls setup.exe.

**Comment by TechTrooper** [comment]
I'm also seeing a lot of "Error unzipping downloaded content" when trying to install packages that perform file and folder copy operations.

I'm using PSAppDeployToolkit installers and they work perfectly when executed manually.

**Comment by okieselbach** [comment]
maybe it was not just a problem of folder structure but with filenames. In that case you might got it fixed now with the latest update of the tool:

Add support for Unicode file and folder name
https://github.com/Microsoft/Microsoft-Win32-Content-Prep-Tool/releases/tag/v1.7

**Comment by nikyusof** [comment]
@sebawern, are you still having problems using the latest version of the tool?

**Comment by sebawern** [comment]
I will have to try again, but I currently do not have any packages that require nested folders in the source.

**Comment by Orava2** [comment]
> 
> 
> Also seeing a lot of "Error unzipping downloaded content" in Intune when trying to install packages that perform file and folder copy operations.
> 
> I'm using PSAppDeployToolkit installers and they work perfectly when executed manually or with SCCM.

I tried to use a batch file which had a xcopy command. I got a unzipping failed error. I needed to remove the xcopy command and unzip the files from a zip archive.

This was the line which caused the unzipping failed error.
`xcopy android-studio "C:\Program Files\android-studio\" /E /Y`

Folder android-studio was taken from a file android-studio-ide-201.7042882-windows.zip (https://developer.android.com/studio#downloads)

Content Prep Tool version is  1.8.2.
